lib/runner.js

```diff
@@ -388,6 +395,13 @@ Runner.prototype.hook = function(name, fn) {
             });
             // a pending hook won't be executed twice.
             hook.pending = true;
+            if (name === HOOK_TYPE_AFTER_ALL) {
```

Comment:
Possible given L380?

Reply:
Yes, it's possible. This statement is part of the `else` block which handles the `before` and `after` hooks.

Reply:
Sorry. Misread hook name condition. But could just do the deprecation check for both `after` hook types prior to L380.
```js
       if (err instanceof Pending) {
         // Skipping after test has run is nonsensical...
         if (name === HOOK_TYPE_AFTER_EACH || name === HOOK_TYPE_AFTER_ALL) {
            utils.deprecate(
              'this.skip() in ' +
                dquote(name + '()') +
                ' hooks is DEPRECATED and will be removed from a future version of Mocha.'
            );
         }
```